### DISCUSSION TITLE
Chapter 4 error

### DISCUSSION THREAD

**Discussion by infinite969** [Cyfrin/foundry-full-course-cu#837] [Q&A] [answered]
Hello frens I am having the following error in my FundMe contracts and AI tools aren't quite helping me figure out the problem.
Note: This contract is running on the sepolia remix VM.

This is the error:

TypeError: Member "getConversionRate" not found or not visible after argument-dependent lookup in uint256.
  --> Fundme.sol:23:17:
   |
23 |         require(msg.value.getConversionRate() >= mininumUSD,"didnt send enough eth"); // This requires the users donation to be greater than 5 USD
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^

Here is my code so far:

/ SPDX-License-Identifier: MIT
 
pragma solidity 0.8.18;
 // This imoprts Chainlink AggregatorV3 Intetrface for retriving price Data
import "@chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";
import {PriceConverter} from "./PriceConverter.sol";
    
// ** Code instuctions **
// Recieve donation from users in ETH
// Withdraw donations
// Keep a list of donation addresses

// ETH/USD PRICE FEED ADDRESS: 0x694AA1769357215DE4FAC081bf1f309aDC325306

contract FundMe {
    using PriceConverter for uint256;
    uint256 public constant mininumUSD  = 5e18; // Public state variable set equal to 5.000000000000000000
    
    address[] public Funders;
    mapping (address funder => uint256 amountedFunded) public addressToAmountFunded;

    function fund() public payable  {
        require(msg.value.getConversionRate() >= mininumUSD,"didnt send enough eth"); // This requires the users donation to be greater than 5 USD
        Funders.push(msg.sender);
        addressToAmountFunded[msg.sender] += addressToAmountFunded[msg.sender] + msg.value;
    }

    
    function getPrice() public view returns(uint256) {
        AggregatorV3Interface pricefeed = AggregatorV3Interface(0x694AA1769357215DE4FAC081bf1f309aDC325306);
        (, int price, , ,) = pricefeed.latestRoundData();
        return uint256(price * 1e10);
    }
    

    function getConversionRate(uint256 ethAmount ) public view returns (uint256){
        uint256 ethPrice = getPrice();
        uint256 ethAmountInUSD = (ethPrice * ethAmount) / 1e18;
        return ethAmountInUSD;
    
    }   
    
    function GetVersion() public view returns (uint256)  {
        return AggregatorV3Interface(0x694AA1769357215DE4FAC081bf1f309aDC325306).version();
    }
}

**Comment by GenevaRoc** [comment, marked as answer]
try using this in the fund funtion instead
`require(getConversionRate(msg.value) >= minimumUsd, "didn't send enough cash");`

**Reply by infinite969** [reply]
this worked! Can you tell me where I was off? Still figuring this out haha

**Reply by GenevaRoc** [reply]
this is from your code 
`require(msg.value.getConversionRate()`

**Reply by GenevaRoc** [reply]
You ought to tell solidity to get conversion rate first then check to see that value/amount being sent is equal to or greater than minimumUsd
